src/server/src/main/java/io/cassandrareaper/service/SegmentRunner.java

```diff
@@ -1146,6 +1180,58 @@ long intensityBasedDelayMillis(double intensity) {
     }
   }
 
+  private boolean lockSegmentRunners() {
+    if (this.repairUnit.getIncrementalRepair()) {
+      return true;
+    }
+
+    try (Timer.Context cx
+        = context.metricRegistry.timer(MetricRegistry.name(SegmentRunner.class, "lockSegmentRunners")).time()) {
+
+      boolean result = context.storage instanceof IDistributedStorage
+          ? ((IDistributedStorage) context.storage).takeLead(repairRunner.getRepairRunId(), LOCK_DURATION)
```

Comment:
Instead of using `repairRunner.getRepairRunId()`, let's use a unique constant UUID that would be declared as `final static` member of SegmentRunner (like `13814000-1dd2-11b2-8080-808080808080` for example, any valid timeuuid would work here I guess).

This will allow to have a global lock which is what we want, not a repair level lock.

Thi

Reply:
I'm left a bit uncertain on this.

The work in this PR is to add validation onto whether we need the locking at all. I suspect that we can't 100% avoid the situation of duplicate segment repairs no matter what we do, and that the opportunistic provided here will do the job in all production environments.

Can we focus this PR on just adding the opportunistic check and leave any corrections to the LWT separate?